connectivity/check/context.go

```diff
@@ -366,6 +372,75 @@ func (ct *ConnectivityTest) skip(t *Test) {
 	t.skipped = true
 }
 
+func (ct *ConnectivityTest) writeJunit() error {
+	if ct.Params().JunitFile == "" {
+		return nil
+	}
+
+	suite := &junit.TestSuite{
+		Name:    "connectivity test",
+		Package: "cilium",
+		Tests:   len(ct.tests),
+		Properties: &junit.Properties{
+			Properties: []junit.Property{
+				{Name: "Args", Value: strings.Join(os.Args[3:], "|")},
+			},
+		},
+	}
+
+	for i, t := range ct.tests {
+		test := &junit.TestCase{
+			Name:      t.Name(),
+			Classname: "connectivity test",
+			Status:    "passed",
+			Time:      t.completionTime.Sub(t.startTime).Seconds(),
+		}
+
+		// Timestamp of the TestSuite is the first test's start time
+		if i == 0 {
+			suite.Timestamp = t.startTime.Format("2006-01-02T15:04:05")
+		}
+		suite.Time += test.Time
+
+		if t.skipped {
+			test.Status = "skipped"
+			test.Skipped = &junit.Skipped{Message: t.Name() + " skipped"}
+			suite.Skipped++
+			test.Time = 0
+		} else if t.failed {
+			test.Status = "failed"
+			test.Failure = &junit.Failure{Message: t.Name() + " failed", Type: "failure"}
+			suite.Failures++
+			msgs := []string{}
+			for _, a := range t.failedActions() {
+				msgs = append(msgs, a.String())
+			}
+			test.Failure.Value = strings.Join(msgs, "\n")
+		}
+
+		suite.TestCases = append(suite.TestCases, test)
+	}
+
+	suites := junit.TestSuites{
+		Tests:      suite.Tests,
+		Disabled:   suite.Skipped,
+		Failures:   suite.Failures,
+		Time:       suite.Time,
+		TestSuites: []*junit.TestSuite{suite},
+	}
+
+	f, err := os.Create(ct.Params().JunitFile)
+	if err != nil {
+		return err
+	}
+
+	if err := suites.WriteReport(f); err != nil {
+		return err
+	}
+
+	return f.Close()
```

Comment:
File won't be closed if `WriteReport` returns an error.
What do you think about closing the file in the `defer` block:
```
	f, err := os.Create(ct.Params().JunitFile)
	if err != nil {
		return err
	}
	defer func() {
		if err := f.Close(); err != nil {
			//log error here
		}
	}()

	return suites.WriteReport(f)
```